### PR TITLE
Implements pagination on the availability page.

### PR DESCRIPTION
**Pull request by meagancheese**
Allows the user to schedule their availability for a month (4 weeks) ahead of the current day.

Screenshot 2020-07-27 at 4 26 00 PM

### CODE REVIEW

File: src/main/webapp/availability.html [hunk 1]
```diff
@@ -39,10 +39,16 @@
...
       <h1 class="text-center">Availability</h1>
-      <h5 class="text-center lead">Update your availability to be an Interviewer for this week below:</h5>
+      <h5 class="text-center lead">
+        Update your availability to be an Interviewer for the week below. Remember that an interview
+        lasts for an hour.
+      </h5>
+      <br>
+      <div id="page-buttons">
+        <button id="back-button" class="btn btn-primary mb-2 btn-sm" onclick="goBack()">&lt Previous Week</button>
+        <button id="forward-button" class="btn btn-primary mb-2 btn-sm" onclick="goForward()">Next Week &gt</button>
```
Comment (apluscs) [inline src/main/webapp/availability.html]: Can we get these a different style from the "submit" buttons we have everywhere else?
https://getbootstrap.com/docs/4.0/components/buttons/
l'm a fan of 
`<button type="button" class="btn btn-light">Light</button>`
or 
one of the outline buttons.

File: src/main/webapp/availability.js [hunk 2]
```diff
@@ -66,3 +66,21 @@ function updateAvailability() {
   let request = new Request('/availability', {method: 'PUT', body: requestBody});
   fetch(request).then(unused => {loadAvailabilityTable(availabilityTableDiv(), browserTimezoneOffset())});
 }
+
+let page = 0;
+
+function goBack() {
+  if (page === 0) {
```
Comment (apluscs) [inline src/main/webapp/availability.js]: l think people can modify the JS so just to be safe put page <== 0, same for the 3 in goForward

File: src/main/webapp/availabilityTable.jsp [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
-<%@ page import="java.util.List,java.time.Instant" %>
+<%@ page import="java.util.List" %>
+<%@ page import="java.time.Instant" %>
+<%@ page import="java.time.temporal.ChronoUnit" %>
 <%@ page import="java.lang.Integer" %>
 <%
   List<List<AvailabilityTimeSlot>> list = AvailabilityTimeSlotGenerator
-    .timeSlotsForWeek(Instant.now(), Integer.parseInt(request
+    .timeSlotsForWeek(Instant.now().plus((Integer.parseInt(request
+    .getParameter("page")) * 7), ChronoUnit.DAYS), Integer.parseInt(request
```
Comment (apluscs) [inline src/main/webapp/availabilityTable.jsp]: Nit: This needs to be indented twice. Same for every continuing line .

**Review by apluscs — CHANGES_REQUESTED** [review]
Small changes

File: src/main/webapp/availability.html (hunk 1 above)
Comment (meagancheese) [reply src/main/webapp/availability.html]: I thought the light was too light / hard to read so I went with a dark grey.

File: src/main/webapp/availability.js (hunk 2 above)
Comment (meagancheese) [reply src/main/webapp/availability.js]: I put some catches for where page is being used throughout.

File: src/main/webapp/availabilityTable.jsp (hunk 3 above)
Comment (meagancheese) [reply src/main/webapp/availabilityTable.jsp]: Done

File: src/main/webapp/availability.js [hunk 4]
```diff
@@ -66,3 +72,29 @@ function updateAvailability() {
...
+}
+
+function goForward() {
+  if (page > 3) {
+    page = 3;
+    return;
+  }
+  if (page === 3) {
+    return;
+  }
```
Comment (apluscs) [inline src/main/webapp/availability.js]: Can shorten this to 
`
if (page >= 3) {
    page = 3;
    return;
  }
`
and similarly for goBack

**Review by apluscs — APPROVED** [review]

File: src/main/webapp/availability.js (hunk 4 above)
Comment (meagancheese) [reply src/main/webapp/availability.js]: Done

**Review by GrantComm — APPROVED** [review]

File: src/main/webapp/availability.js [hunk 5]
```diff
@@ -33,7 +33,13 @@ function toggleTile(tile) {
 }
 
 function loadAvailabilityTable(tableDiv, timezoneOffset) {
-  fetch(`/availabilityTable.jsp?timeZoneOffset=${timezoneOffset}`)
+  if (page < 0) {
```
Comment (acmcarther) [inline src/main/webapp/availability.js]: You've got guard logic for this in `goForward` and `goBack`, so this should not be necessary

File: src/main/webapp/availability.js [hunk 6]
```diff
@@ -66,3 +72,23 @@ function updateAvailability() {
...
+  if (page <= 0) {
+    page = 0;
+    return;
+  }
+  page -= 1;
+  loadAvailabilityTable(availabilityTableDiv(), browserTimezoneOffset());
+}
+
+function goForward() {
+  if (page >= 3) {
```
Comment (acmcarther) [inline src/main/webapp/availability.js]: Can you move this `3` to a constant somewhere?

File: src/main/webapp/availabilityTable.jsp [hunk 7]
```diff
@@ -1,12 +1,15 @@
...
+<%@ page import="java.util.List" %>
+<%@ page import="java.time.Instant" %>
+<%@ page import="java.time.temporal.ChronoUnit" %>
 <%@ page import="java.lang.Integer" %>
 <%
   List<List<AvailabilityTimeSlot>> list = AvailabilityTimeSlotGenerator
-    .timeSlotsForWeek(Instant.now(), Integer.parseInt(request
-    .getParameter("timeZoneOffset")), new DatastoreAvailabilityDao());
+      .timeSlotsForWeek(Instant.now().plus((Integer.parseInt(request
+      .getParameter("page")) * 7), ChronoUnit.DAYS), Integer.parseInt(request
```
Comment (acmcarther) [inline src/main/webapp/availabilityTable.jsp]: Fix indentation.

**Review by acmcarther — CHANGES_REQUESTED** [review]

File: src/main/webapp/availabilityTable.jsp (hunk 7 above)
Comment (meagancheese) [reply src/main/webapp/availabilityTable.jsp]: Claire was asking for this to be indented twice because it's the rest of the original line. Should it only be indented once?

File: src/main/webapp/availability.js (hunk 5 above)
Comment (meagancheese) [reply src/main/webapp/availability.js]: Done

File: src/main/webapp/availability.js (hunk 6 above)
Comment (meagancheese) [reply src/main/webapp/availability.js]: Done

File: src/main/webapp/availabilityTable.jsp (hunk 7 above)
Comment (acmcarther) [reply src/main/webapp/availabilityTable.jsp]: These should not be aligned:
```java
      .timeSlotsForWeek(Instant.now().plus((Integer.parseInt(request
      .getParameter("page")) * 7), ChronoUnit.DAYS), Integer.parseInt(request
```

They're at different levels of nesting. The same applies below. The alignment here is far from helpful -- it's actually actively misleading.

Here's what this should probably look like:

```java
  List<List<AvailabilityTimeSlot>> list = 
    AvailabilityTimeSlotGenerator.timeSlotsForWeek(
        Instant.now().plus(
            7 * Integer.parseInt(request.getParameter("page")), 
            ChronoUnit.DAYS), 
        Integer.parseInt(request.getParameter("timeZoneOffset")), 
        new DatastoreAvailabilityDao());
```

The significant part is that the line breaks happen between parameters. All levels of nesting are consistent within the argument set that they are a part of.

That said, by the way, if any of those `parse ints` throw you'd be causing a headache for someone to debug. Generally I support inlining, but when the things being inlined can throw their own exceptions (and there are several of them), then you can cause a bit of code to become a hassle to debug.

Comment (meagancheese) [reply src/main/webapp/availabilityTable.jsp]: Handled.

**Comment by meagancheese** [comment]
This is ready for another review.

File: src/main/webapp/availabilityTable.jsp [hunk 8]
```diff
@@ -26,7 +32,7 @@
       <tr>
         <c:forEach items = "${pageScope.list}" var = "day">
           <td onclick="toggleTile(this)" data-utc="${day.get(i).utcEncoding()}" 
-            class="${day.get(i).selected() ? (day.get(i).scheduled() ? 
+              class="${day.get(i).selected() ? (day.get(i).scheduled() ?
```
Comment (acmcarther) [inline src/main/webapp/availabilityTable.jsp]: I know this was how the code originally was, but chained ternary is pretty unreadable. Can this be refactored (maybe calculate this in the java code above?)

**Review by acmcarther — APPROVED** [review]
